Splits I/O JSON Schema

glacials opened the pull request:
Adds a canonical Splits I/O JSON schema specified by [json-schema][1].

[1]: http://json-schema.org/

glacials commented on public/schema/README.md:
```diff
@@ -0,0 +1,9 @@
+
+[**Raw JSON Schema**][2]
+
+[1]: http://lbovet.github.io/docson/index.html#https://raw.githubusercontent.com/glacials/splits-io/master/public/schema/run_v1.0.0.json
```
This link won't work before merging, but you can use http://lbovet.github.io/docson/index.html#https://raw.githubusercontent.com/glacials/splits-io/standard_format/public/schema/run_v1.0.0.json to see what it looks like.

BatedUrGonnaDie commented:
Is the game and category shortname being just one string enough?  Every service has it's own id/shortname for them so how do you decide what becomes the cannonical shortname.

glacials commented:
Good point! We can add a `links` object to games and categories similar to the one we have for runners.

glacials commented:
Added `links` to games and categories, removed empty `required` arrays (apparently that's against json-schema spec), added a barebones example run to the README (included below as well), and stopped requiring array fields (rather than requiring consumers to specify `"runners": []`). 

```json
{
  "_schemaVersion": "v1.0.0",
  "timer": {
    "shortname": "LiveSplit",
    "longname":  "livesplit", 
    "version":   "v1.6.0"
  },
  "game":     {"longname": "Super Mario Odyssey"},
  "category": {"longname": "Any%"},
  "segments": [
    {
      "name":         "Cap",
      "duration":     {"realtimeMS": 143123},
      "bestDuration": {"realtimeMS": 143167}
    }
  ]
}
```

Updated docs generated from this schema are at http://lbovet.github.io/docson/index.html#https://raw.githubusercontent.com/glacials/splits-io/standard_format/public/schema/run_v1.0.0.json

@CryZe / @wooferzfg can you take a look at this, and let me know if this seems suitable for implementation into livesplit-core?
